### ISSUE TITLE
Install Error Hancom Word

### ISSUE THREAD

**Issue by leejongyoung** [Homebrew/homebrew-cask#141562]
### Verification

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [X] I have retried my command with `--force`.
- [X] I ran `brew update-reset && brew update` and retried my command.
- [X] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [X] I made doubly sure this is not a [checksum does not match](https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match) error.

### Description of issue

Downloading Hangul and computer fails.
You can download it normally from the official site.
The file name has not changed or the URL has not changed.

```
$HOME/Library/Caches/Homebrew/downloads
```

I deleted the file and tried again, but the error still remains.

### Command that failed

brew install --cask --verbose --debug hancom-word

### Output of command with `--verbose --debug`

```shell
leejongyoung@MacBook-Air ~ % brew install --cask --verbose --debug hancom-word
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading hancom-word
==> Cask::Installer#install
==> Printing caveats
==> Cask::Installer#fetch
==> Downloading https://cdn.hancom.com/pds/hnc/DOWN/HancomDocs/HwpMac_HancomDocs.pkg
/usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.0.1\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 13.2.1\)\ curl/7.86.0 --header Accept-Language:\ en --retry 3 --location --silent --head --request GET https://cdn.hancom.com/pds/hnc/DOWN/HancomDocs/HwpMac_HancomDocs.pkg
Already downloaded: /Users/leejongyoung/Library/Caches/Homebrew/downloads/3edba80c3369f86447bd6495b1d4f62fb5b653c54b668412154ed93d8374d11c--
==> Checking quarantine support
/usr/bin/env /usr/bin/xattr -h
/usr/bin/env /usr/bin/swift -target arm64-apple-macosx13 /opt/homebrew/Library/Homebrew/cask/utils/quarantine.swift
==> Quarantine is available.
==> Verifying Gatekeeper status of /Users/leejongyoung/Library/Caches/Homebrew/downloads/3edba80c3369f86447bd6495b1d4f62fb5b653c54b668412154ed93d8374d11c--
/usr/bin/env /usr/bin/xattr -p com.apple.quarantine /Users/leejongyoung/Library/Caches/Homebrew/downloads/3edba80c3369f86447bd6495b1d4f62fb5b653c54b668412154ed93d8374d11c--
==> /Users/leejongyoung/Library/Caches/Homebrew/downloads/3edba80c3369f86447bd6495b1d4f62fb5b653c54b668412154ed93d8374d11c-- is quarantined
Warning: No checksum defined for cask 'hancom-word', skipping verification.
/usr/bin/env hdiutil imageinfo -format /Users/leejongyoung/Library/Caches/Homebrew/downloads/3edba80c3369f86447bd6495b1d4f62fb5b653c54b668412154ed93d8374d11c--
==> Installing Cask hancom-word
==> Cask::Installer#stage
==> Extracting primary container
==> Using container class UnpackStrategy::Uncompressed for /Users/leejongyoung/Library/Caches/Homebrew/downloads/3edba80c3369f86447bd6495b1d4f62fb5b653c54b668412154ed93d8374d11c--
cp -p /Users/leejongyoung/Library/Caches/Homebrew/downloads/3edba80c3369f86447bd6495b1d4f62fb5b653c54b668412154ed93d8374d11c-- /opt/homebrew/Caskroom/hancom-word/12.30.0,4281
==> Purging files for version 12.30.0,4281 of Cask hancom-word
Error: Is a directory - read
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:1388:in `copy_stream'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:1388:in `block (2 levels) in copy_file'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:1387:in `open'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:1387:in `block in copy_file'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:1386:in `open'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:1386:in `copy_file'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:492:in `copy_file'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:419:in `block in cp'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:1558:in `block in fu_each_src_dest'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:1572:in `fu_each_src_dest0'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:1556:in `fu_each_src_dest'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:418:in `cp'
/opt/homebrew/Library/Homebrew/unpack_strategy/uncompressed.rb:19:in `extract_to_dir'
/opt/homebrew/Library/Homebrew/unpack_strategy.rb:142:in `extract'
/opt/homebrew/Library/Homebrew/unpack_strategy/uncompressed.rb:12:in `extract_nestedly'
/opt/homebrew/Library/Homebrew/cask/installer.rb:220:in `extract_primary_container'
/opt/homebrew/Library/Homebrew/cask/installer.rb:79:in `stage'
/opt/homebrew/Library/Homebrew/cask/installer.rb:107:in `install'
/opt/homebrew/Library/Homebrew/cask/cmd/install.rb:110:in `block in install_casks'
/opt/homebrew/Library/Homebrew/cask/cmd/install.rb:109:in `each'
/opt/homebrew/Library/Homebrew/cask/cmd/install.rb:109:in `install_casks'
/opt/homebrew/Library/Homebrew/cmd/install.rb:185:in `install'
/opt/homebrew/Library/Homebrew/brew.rb:93:in `<main>'
leejongyoung@MacBook-Air ~ %
```

### Output of `brew doctor` and `brew config`

```shell
Your system is ready to brew.

leejongyoung@MacBook-Air ~ % brew config
HOMEBREW_VERSION: 4.0.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: 17c872fb5275d87922a56416587cb439a5064354
Last commit: 4 days ago
Core tap JSON: 20 Feb 02:47 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew tap`

```shell
homebrew/cask
```

**Comment by miccal** [comment]
Cannot reproduce:
```
|-> brew install hancom-word
==> Downloading https://cdn.hancom.com/pds/hnc/DOWN/HancomDocs/HwpMac_HancomDocs.pkg
######################################################################## 100.0%
Warning: No checksum defined for cask 'hancom-word', skipping verification.
==> Installing Cask hancom-word
Warning: macOS's Gatekeeper has been disabled for this Cask
==> Running installer for hancom-word; your password may be necessary.
Package installers may write to any location; options such as `--appdir` are ignored.
Password:
installer: Package name is 
installer: Installing at base path /
installer: The install was successful.
🍺  hancom-word was successfully installed!
```
```
|-> brew config
HOMEBREW_VERSION: 4.0.1-62-ga222a5b
ORIGIN: https://github.com/Homebrew/brew
HEAD: a222a5bbc83923e4c9c09ed0ff07dfe64cf7d863
Last commit: 15 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ceb6f460128d3233b2f40e4d4d1be885465b965e
Core tap last commit: 7 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_AUTOREMOVE: set
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_DEVELOPER: set
HOMEBREW_DISPLAY_INSTALL_TIMES: set
HOMEBREW_EDITOR: /usr/bin/open -e -W
HOMEBREW_FORCE_BREWED_CA_CERTIFICATES: set
HOMEBREW_FORCE_BREWED_CURL: set
HOMEBREW_FORCE_BREWED_GIT: set
HOMEBREW_FORCE_VENDOR_RUBY: set
HOMEBREW_GIT_EMAIL: miccal.matthews@gmail.com
HOMEBREW_GIT_NAME: miccal
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_INSTALL_FROM_API: set
HOMEBREW_UPDATE_REPORT_ALL_FORMULAE: set
Homebrew Ruby: 2.6.8 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 14.0.0 build 1400
Git: 2.39.2 => /opt/homebrew/opt/git/bin/git
Curl: 7.88.0 => /opt/homebrew/opt/curl/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```

**Comment by miccal** [comment]
Also cannot reproduce on my Intel machine:
```
|-> brew config
HOMEBREW_VERSION: 4.0.1-62-ga222a5b
ORIGIN: https://github.com/Homebrew/brew
HEAD: a222a5bbc83923e4c9c09ed0ff07dfe64cf7d863
Last commit: 15 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ceb6f460128d3233b2f40e4d4d1be885465b965e
Core tap last commit: 13 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_AUTOREMOVE: set
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_DEVELOPER: set
HOMEBREW_DISPLAY_INSTALL_TIMES: set
HOMEBREW_EDITOR: /usr/bin/open -e -W
HOMEBREW_FORCE_BREWED_CA_CERTIFICATES: set
HOMEBREW_FORCE_BREWED_CURL: set
HOMEBREW_FORCE_BREWED_GIT: set
HOMEBREW_FORCE_VENDOR_RUBY: set
HOMEBREW_GIT_EMAIL: miccal.matthews@gmail.com
HOMEBREW_GIT_NAME: miccal
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_INSTALL_FROM_API: set
HOMEBREW_UPDATE_REPORT_ALL_FORMULAE: set
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 14.0.0 build 1400
Git: 2.39.2 => /usr/local/opt/git/bin/git
Curl: 7.88.0 => /usr/local/opt/curl/bin/curl
macOS: 13.2.1-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

**Comment by miccal** [comment]
Possibly API-related?

**Comment by leejongyoung** [comment]
I don't know if it's an API problem, but I checked that the same error occurred from another user.

![KakaoTalk_Photo_2023-02-20-15-52-55](https://user-images.githubusercontent.com/105860712/220034336-adced04d-b6aa-4b91-afe9-97fe5cd7c2a2.jpeg)

It may be a temporary problem, so if the error is continuously confirmed, I will register the issue again.

Thank you for confirming.

**Comment by bevanjkay** [comment]
My best guess is that the "referer" is not passed through when fetching the file, when using the API.

**Comment by leejongyoung** [comment]
``` sh
$ export HOMEBREW_NO_INSTALL_FROM_API=1
$ brew install --cask hancom-word
```

```
==> Downloading https://cdn.hancom.com/pds/hnc/DOWN/HancomDocs/HwpMac_HancomDocs.pkg
######################################################################## 100.0%
Warning: No checksum defined for cask 'hancom-word', skipping verification.
==> Installing Cask hancom-word
==> Running installer for hancom-word; your password may be necessary.
Package installers may write to any location; options such as `--appdir` are ignored.
Password:
installer: Package name is 
installer: Installing at base path /
installer: The install was successful.
🍺  hancom-word was successfully installed!
```

If you run the command and proceed with the installation, it will be temporarily resolved.

**Comment by bevanjkay** [comment]
Thanks @leejongyoung 

URL parameters are not currently being passed due to them not being available through the API.
You can temporarily workaround this by setting the environment variable `HOMEBREW_NO_INSTALL_FROM_API=1` but note that this will disable the new API functionality of Homebrew and revert to using git instead, and should not be considered the long term solution. The bug is being tracked here - https://github.com/Homebrew/brew/issues/14728 - and once fixed you can resume using the API by unsetting the variable.

**Comment by Bo98** [comment]
I believe I have fixed this and it will be available with Homebrew 4.0.4.

**Comment by leejongyoung** [comment]
It seems that this problem has been fixed in v4.0.4
```
leejongyoung@ijongyecBookAir ~ % brew install --cask hancom-word
==> Installing Cask hancom-word
==> Running installer for hancom-word; your password may be necessary.
Package installers may write to any location; options such as `--appdir` are ignored.
Password:
installer: Package name is 
installer: Installing at base path /
installer: The install was successful.
🍺  hancom-word was successfully installed!
```

```
leejongyoung@ijongyecBookAir ~ % brew config
HOMEBREW_VERSION: 4.0.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4d0154fa7926e8a7b06e9b074dd45957e820407c
Last commit: 3 days ago
Core tap JSON: 01 Mar 07:18 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.39.2 => /opt/homebrew/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```
